lib/ansible/modules/files/find.py

```diff
@@ -230,31 +231,31 @@ def pfilter(f, patterns=None, excludes=None, use_regex=False):
     if use_regex:
         if patterns and excludes is None:
             for p in patterns:
-                r = re.compile(p)
+                r = re.compile(to_bytes(p, errors='surrogate_or_strict'))
                 if r.match(f):
                     return True
 
         elif patterns and excludes:
             for p in patterns:
-                r = re.compile(p)
+                r = re.compile(to_bytes(p, errors='surrogate_or_strict'))
                 if r.match(f):
                     for e in excludes:
-                        r = re.compile(e)
+                        r = re.compile(to_bytes(e, errors='surrogate_or_strict'))
                         if r.match(f):
```

Comment:
```f``` should be renamed.... Maybe ```b_filepath``` to address both the prefix and the obscure variable naming.  Would be good to docstring that this takes a byte string for filepath.